File: src/main/java/icu/samnyan/aqua/sega/maimai/handler/impl/UserLoginHandler.java
package icu.samnyan.aqua.sega.maimai.handler.impl;

import com.fasterxml.jackson.core.JsonProcessingException;
import icu.samnyan.aqua.sega.general.BaseHandler;
import icu.samnyan.aqua.sega.maimai.model.response.UserLoginResp;
import icu.samnyan.aqua.sega.util.jackson.BasicMapper;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import java.util.Map;

/**
 * @author samnyan (privateamusement@protonmail.com)
 */
@Component("MaimaiUserLoginHandler")
public class UserLoginHandler implements BaseHandler {

    private static final Logger logger = LoggerFactory.getLogger(UserLoginHandler.class);

    private final BasicMapper mapper;

    public UserLoginHandler(BasicMapper mapper) {
        this.mapper = mapper;
    }

    @Override
    public String handle(Map<String, Object> request) throws JsonProcessingException {
        String userId = (String) request.get("userId");

        UserLoginResp resp = new UserLoginResp();

        String json = mapper.write(resp);
        logger.info("Response: " + json);
        return json;
    }
}

File: src/main/java/icu/samnyan/aqua/sega/maimai2/handler/UserLoginHandler.kt
package icu.samnyan.aqua.sega.maimai2.handler

import icu.samnyan.aqua.sega.general.BaseHandler
import icu.samnyan.aqua.sega.util.jackson.BasicMapper
import org.springframework.stereotype.Component

/**
 * @author samnyan (privateamusement@protonmail.com)
 */
@Component("Maimai2UserLoginHandler")
class UserLoginHandler(mapper: BasicMapper) : BaseHandler {
    val resp = mapper.write(mapOf(
        "returnCode" to 1,
        "lastLoginDate" to "2020-01-01 00:00:00.0",
        "loginCount" to 1,
        "consecutiveLoginCount" to 0,
        "loginId" to 1,
        "Bearer" to "meow" // Isn't actually used by the game
    ))

    override fun handle(request: Map<String, Any>) = resp
}
